tests/onnx/test_onnx_model_export.py

```diff
@@ -39,94 +46,86 @@ def data():
-    model.compile(loss="mean_squared_error", optimizer="SGD")
-    model.fit(x, y)
-    return model
-
-
-@pytest.fixture(scope="module")
-def onnx_model(model):
-    import onnxmltools
-
-    return onnxmltools.convert_keras(model)
+    dataset = [(xi.astype(np.float32), yi.astype(np.float32)) for xi, yi in zip(x.values, y.values)]
+    return dataset
 
 
 @pytest.fixture(scope="module")
-def sklearn_model(data):
-    from sklearn.linear_model import LogisticRegression
-
-    x, y = data
-    model = LogisticRegression()
-    model.fit(x, y)
-    return model
+def sample_input(dataset):
+    dataloader = DataLoader(dataset, batch_size=5, num_workers=1,)
+    # Load a batch from the data loader and return the samples
+    x, _ = next(iter(dataloader))
+    return x
 
 
 @pytest.fixture(scope="module")
-def onnx_sklearn_model(sklearn_model):
-    import onnxmltools
-    from skl2onnx.common.data_types import FloatTensorType
-
-    initial_type = [("float_input", FloatTensorType([None, 4]))]
-    onx = onnxmltools.convert_sklearn(sklearn_model, initial_types=initial_type)
-    return onx
+def model(dataset):
+    model = nn.Sequential(nn.Linear(4, 3), nn.ReLU(), nn.Linear(3, 1),)
+    criterion = nn.MSELoss()
+    optimizer = torch.optim.SGD(model.parameters(), lr=0.01)
+    batch_size = 16
+    num_workers = 4
+    dataloader = DataLoader(
+        dataset, batch_size=batch_size, num_workers=num_workers, shuffle=True, drop_last=False
+    )
 
+    model.train()
+    for _ in range(5):
+        for batch in dataloader:
+            optimizer.zero_grad()
+            batch_size = batch[0].shape[0]
+            y_pred = model(batch[0]).squeeze(dim=1)
+            loss = criterion(y_pred, batch[1])
+            loss.backward()
+            optimizer.step()
 
-@pytest.fixture(scope="module")
-def predicted(model, data):
-    return model.predict(data[0])
+    return model
 
 
-@pytest.fixture(scope="module")
-def tf_model_multiple_inputs_float64():
-    graph = tf.Graph()
-    with graph.as_default():
-        t_in1 = tf.placeholder(tf.float64, 10, name="first_input")
-        t_in2 = tf.placeholder(tf.float64, 10, name="second_input")
-        t_out = tf.multiply(t_in1, t_in2)
-        tf.identity(t_out, name="output")
-    return graph
+@pytest.fixture
+def onnx_model(model, sample_input, tmpdir):
+    model_path = os.path.join(str(tmpdir), "torch_onnx")
+    dynamic_axes = {"input": {0: "batch"}}
+    torch.onnx.export(
+        model, sample_input, model_path, dynamic_axes=dynamic_axes, input_names=["input"]
+    )
+    return onnx.load(model_path)
 
 
 @pytest.fixture(scope="module")
-def tf_model_multiple_inputs_float32():
-    graph = tf.Graph()
-    with graph.as_default():
-        t_in1 = tf.placeholder(tf.float32, 10, name="first_input")
-        t_in2 = tf.placeholder(tf.float32, 10, name="second_input")
-        t_out = tf.multiply(t_in1, t_in2)
-        tf.identity(t_out, name="output")
-    return graph
+def onnx_sklearn_model():
```

Comment:
If [this comment](https://github.com/mlflow/mlflow/pull/3844/files?file-filters%5B%5D=.py#r543074478) is actually what you want to test, then why are you testing `onnx_sklearn_model `. Not the pytorch run